s2n.mk

```diff
@@ -116,6 +116,10 @@ ifdef S2N_DEBUG
 	CFLAGS += ${DEBUG_CFLAGS}
 endif
 
+ifdef S2N_GDB
+	CFLAGS += ${DEBUG_CFLAGS} -Og
+endif
```

Comment:
Instead of duplicating the work of S2N_DEBUG, how about defining S2N_DEBUG if S2N_GDB is defined (before the rest of the makefile executes) and then also appending -0g at the end? That way, if we make S2N_DEBUG behavior changes, S2N_GDB inherits them.